presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1578,6 +1588,13 @@ protected Scope visitSampledRelation(SampledRelation relation, Optional<Scope> s
             if (!(samplePercentageObject instanceof Number)) {
                 throw new SemanticException(NON_NUMERIC_SAMPLE_PERCENTAGE, relation.getSamplePercentage(), "Sample percentage should evaluate to a numeric expression");
             }
+            Expression samplePercentage = relation.getSamplePercentage();
```

Comment:
Instead of this approach, I think it would be better to take advantage of the type coercion that Presto provides.  I can't put my comment there, but at line 1582-1584, I would call `evaluateConstantExpression` (static method from ExpressionInterpreter) that takes an expected type, and will add the appropriate casts if necessary.  the downside is it will throw a different error code if it cannot be cast to a double (TYPE_MISMATCH semantic exception, with a message that it cannot cast from type x to double).  I think that's actually fine, but we can always catch and rethrow the exception to be the NON_NUMERIC_SAMPLE_PERCENTAGE exception

Reply:
Done! Thanks